sympy/polys/matrices/ddm.py

```diff
@@ -598,22 +598,55 @@ def rref_den(a):
         return b, denom, pivots
 
     def nullspace(a):
+        """Returns a basis for the nullspace of a.
+
+        The domain of the matrix must be a field.
+
+        See Also
+        ========
+
+        rref
+        sympy.polys.matrices.domainmatrix.DomainMatrix.nullspace
+        """
         rref, pivots = a.rref()
-        rows, cols = a.shape
-        domain = a.domain
+        return rref.nullspace_from_rref(pivots)
```

Comment:
Why does this have to be different method?

Reply:
It was originally so that you can compute the nullspace and particular solutions from the RREF without needing to compute the RREF twice:
https://github.com/sympy/sympy/blob/12d7761b5033628cc12dc2f3b3df9546c2378086/sympy/polys/matrices/linsolve.py#L92-L97
That is used by `linsolve` to generate solutions when there is not a unique solution.

Reply:
Also there are two `rref` methods: `rref` and `rref_den` where the first s inly for fields and the second can be for any domain. The `nullspace_from_rref` methods can be used after either method because it does not require division. Basically there are multiple ways to get to an RREF and then multiple things that you can do when you have an RREF. The `nulllspace_from_rref` does one of the things that can be done with an RREF.

Reply:
> Also there are two `rref` methods: `rref` and `rref_den` where the first s inly for fields and the second can be for any domain.

Actually to be clear `rref_den` requires `exquo` but that is available for all the usual domains in SymPy.